### ISSUE TITLE
Exception when saving Company: Apply email template set

### ISSUE THREAD

**Issue by dallenk** [iomad/iomad#2054]
I attempted to apply an email template set to a company from the dropdown list and see this error.

Exception - iomad::has_capability(): Argument #2 ($context) must be of type context, null given, called in [dirroot]/blocks/iomad_company_admin/company_edit_form.php on line 563

Debug info:
Error code: generalexceptionmessage×Dismiss this notification
Stack trace:
line 1836 of /local/iomad/lib/iomad.php: TypeError thrown
line 563 of /blocks/iomad_company_admin/company_edit_form.php: call to iomad::has_capability()
×Dismiss this notification
Output buffer: <br /> <b>Warning</b>: Undefined variable $companycontext in <b>/var/www/vhosts/sandbox.domain.ca/iomad/blocks/iomad_company_admin/company_edit_form.php</b> on line <b>563</b><br />

**Comment by turf212** [comment]
That's weird.  That line in the current code doesn't match up as line 563 has nothing about capabilities in it

https://github.com/iomad/iomad/blob/99de84b1004d5ca482ff8d0e35ef67afd9019f0b/blocks/iomad_company_admin/company_edit_form.php#L563

The underlying function of this doesn't have any capability checks either.

In the company edit form the companycontext is set in a couple of places

1 - if the company is not new - it is set in line 62:

```
if (!$new) {
    // Set the companyid
    $companyid = iomad::get_my_companyid($systemcontext);
    $companycontext =  \core\context\company::instance($companyid);

```

if the company being created is new then its initially set to the systemcontext (line 99) and if there is a parent company ID being passed, then it is set to that in line 105.

I suspect your local code is out as in the commits on that file from since March 2 2024 - none of the files have a capability check on line 563.

**Comment by dallenk** [comment]
this one is still 402.

```
sudo -u www-data git log --oneline
1610dbd2b68 (HEAD -> IOMAD_402_STABLE_new, origin/IOMAD_402_STABLE) IOMAD: < 4.3 clean up redirect to own company URL on login to the wrong one
1e000b0d860 IOMAD: missing string for capability trainingevent:viewallattendees
cc4091da4f5 IOMAD: Getting error when there are no courses available for manager enrolment as form was counting 'All' as a course.
```

hard reset done as well

**Comment by turf212** [comment]
Ah - well - that does make a difference :)  One f the upstream merges for fixes from 4.3 changed the context check.  Should have been systemcontext not companycontext for the variable.  Pushed a fix.
